### ISSUE TITLE
Cannot extract IP from DBSystem (using example code)

### ISSUE THREAD

**Issue by mauropagano** [oracle/terraform-provider-baremetal#60]
When trying to extract the (private) IP for a DBSystem using the example available here on GitHub I get

* Resource 'data.baremetal_database_db_node.DBNodeDetails' does not have attribute 'vnic_id' for variable 'data.baremetal_database_db_node.DBNodeDetails.vnic_id'

and the error looks legit since even the terraform.tfstate (or "terraform show") has no vnic exposed.

Not sure I'm doing something wrong but 1. terraform itself doesn't show the vnic at all and 2. I'm just using the code provided, makes me wonder if the issue isn't underlying in the provider not properly populating the data

**Comment by noeldcosta** [comment]
Getting the same error. Does anyone have a working example or is this a defect that needs to be fixed.

**Comment by abejunejo** [comment]
Running into the same issue

**Comment by josephholsten** [comment]
I'll take responsibility for this one

**Comment by craigcarl-oracle** [comment]
@josephholsten https://github.com/oracle/terraform-provider-baremetal/tree/master/docs/examples/DBsystems

**Comment by josephholsten** [comment]
@mauropagano sorry for the delay here, I've been trying to get load balancer support ready! I'm going to try to throw together a minimal test case, but if you have one ready that would be handy.

**Comment by mauropagano** [comment]
@josephholsten I sure can try! :-)
I didn't test it completely (aka make sure it reproduces the issue) since it takes about 1h to run and I started it like 10 mins ago, in case it does NOT reproduce I'll update this issue again but it's the same exact code I used when creating this issue the first time so I'm assuming nothing changed.

I split files based on what they define just because they stay "cleaner" imho that way.

data.tf
```
data "baremetal_identity_availability_domains" "ADs" {
  compartment_id = "${var.tenancy_ocid}"
}
data "baremetal_database_db_node" "DBNodeDetails" {
   db_node_id = "${baremetal_database_db_system.TFDBNode.id}"
}
data "baremetal_core_vnic" "DBNodeVnic" {
    vnic_id = "${lookup(data.baremetal_database_db_node.DBNodeDetails.vnic_id,"vnic_id")}"
}
```

output.tf
```
output "instance_id" {
    value = "${baremetal_database_db_system.TFDBNode.id}"
}
output "DBNodePublicIP" {
    value = ["${data.baremetal_core_vnic.DBNodeVnic.public_ip_address}"]
}
output "DBNodePrivateIP" {
    value = ["${data.baremetal_core_vnic.DBNodeVnic.private_ip_address}"]
}
```

provider.tf
```
provider "baremetal" {
  tenancy_ocid = "${var.tenancy_ocid}"
  user_ocid = "${var.user_ocid}"
  fingerprint = "${var.fingerprint}"
  private_key_path = "${var.private_key_path}"
  timeout_minutes = "${var.timeout_minutes}"
}
```

resource.tf (it's JSON since I create it programmatically)
```
{
  "resource": [
    {
      "baremetal_database_db_system": {
        "TFDBNode": {
          "domain": "${var.DBNodeDomainName}",
          "availability_domain": "${var.availability_domain}",
          "display_name": "${var.DBNodeDisplayName}",
          "compartment_id": "${var.compartment_ocid}",
          "subnet_id": "${var.subnet_ocid}",
          "hostname": "${var.DBNodeHostName}",
          "database_edition": "${var.DBEdition}",
          "ssh_public_keys": [
            "${var.ssh_public_key}"
          ],
          "shape": "${var.DBNodeShape}",
          "disk_redundancy": "${var.DBDiskRedundancy}",
          "cpu_core_count": "${var.CPUCoreCount}",
          "db_home": {
            "db_version": "${var.DBVersion}",
            "display_name": "${var.DBDisplayName}",
            "database": {
              "db_name": "${var.DBName}",
              "admin_password": "${var.DBAdminPassword}"
            }
          },
          "lifecycle": {
            "ignore_changes": [
              "db_home.0.database.0.admin_password",
              "db_home.0.database.0.db_name",
              "db_home.0.db_version",
              "db_home.0.display_name"
            ]
          }
        }
      }
    }
  ]
}
```

vars.tf
```
variable "tenancy_ocid" {}
variable "user_ocid" {}
variable "fingerprint" {}
variable "private_key_path" {}
variable "availability_domain" {}
variable "compartment_ocid" {}
variable "subnet_ocid" {}
variable "ssh_public_key" {}
variable "DBNodeShape" {}
variable "CPUCoreCount" {}
variable "DBEdition" {}
variable "DBAdminPassword" {}
variable "DBName" {}
variable "DBVersion" {}
variable "DBDisplayName" {}
variable "DBDiskRedundancy" {}
variable "DBNodeDisplayName" {}
variable "DBNodeDomainName" {}
variable "DBNodeHostName" {}
variable "timeout_minutes" {
    default = 5
}
```

vars.json (I took out a few "private" values)
```
{
  "user_ocid": "",
  "DBEdition": "ENTERPRISE_EDITION_EXTREME_PERFORMANCE",
  "availability_domain": "foTc:PHX-AD-1",
  "subnet_ocid": "",
  "compartment_ocid": "",
  "tenancy_ocid": "",
  "DBDiskRedundancy": "HIGH",
  "DBNodeDomainName": "enkitec",
  "DBNodeHostName": "node1",
  "CPUCoreCount": "2",
  "DBNodeDisplayName": "RandomDBSystem",
  "DBAdminPassword": "",
  "ssh_public_key": "",
  "DBVersion": "12.1.0.2",
  "fingerprint": "",
  "private_key_path": "auth_key",
  "DBNodeShape": "BM.DenseIO1.36",
  "DBDisplayName": "DBfromAPI",
  "DBName": "TESTBMC"
}
```

Just pass inline the variable file (terraform apply -var-file=vars.json)

I hope it helps (and that it gets formatted decently by GitHub!)

**Comment by josephholsten** [comment]
awesome! I'll start using that immediately

**Comment by mauropagano** [comment]
Testcase worked on my side (in the sense it reproduces the error), hopefully reproduced on your side too!

**Comment by josephholsten** [comment]
@mauropagano it seems there is a bug with the example. We'll be adding a simple way to get the ip address with #94, and update the example once that's merged.

**Comment by mauropagano** [comment]
Thanks! 
Quick question (maybe too soon to ask but I figure I'd try), will the syntax / approach change for baremetal_core_instance too or just for baremetal_database_db_system?

**Comment by craigcarl-oracle** [comment]
@josephholsten @codycushing Were DBSystems included in #94 or just compute. If we're adding this to DBSystems (I like the consistency) we should discuss what we're returning for RAC and Exadata.

**Comment by mikejihbe** [comment]
I think there is actually a bug in your example. The `baremetal_database_db_node` datasource takes a db_node_id, not a db_system_id. This is a fully working example that stores `vnic_id` on the state for the db_node. The key differentiators you're looking for are on the very bottom of this example. 

```
variable "ssh_public_key" {
    default = "ssh-rsa KKKLK3NzaC1yc2EAAAADAQABAAABAQC+UC9MFNA55NIVtKPIBCNw7++ACXhD0hx+Zyj25JfHykjz/QU3Q5FAU3DxDbVXyubgXfb/GJnrKRY8O4QDdvnZZRvQFFEOaApThAmCAM5MuFUIHdFvlqP+0W+ZQnmtDhwVe2NCfcmOrMuaPEgOKO3DOW6I/qOOdO691Xe2S9NgT9HhN0ZfFtEODVgvYulgXuCCXsJs+NUqcHAOxxFUmwkbPvYi0P0e2DT8JKeiOOC8VKUEgvVx+GKmqasm+Y6zHFW7vv3g2GstE1aRs3mttHRoC/JPM86PRyIxeWXEMzyG5wHqUu4XZpDbnWNxi6ugxnAGiL3CrIFdCgRNgHz5qS1l MustWin"
}
variable "compartment_id" {}

provider "baremetal" {
  tenancy_ocid = "${var.tenancy_ocid}"
  user_ocid = "${var.user_ocid}"
  fingerprint = "${var.fingerprint}"
  private_key_path = "${var.private_key_path}"

}


data "baremetal_identity_availability_domains" "ADs" {
  compartment_id = "${var.compartment_id}"
}

resource "baremetal_core_virtual_network" "t" {
	cidr_block = "10.0.0.0/16"
	compartment_id = "${var.compartment_id}"
	display_name = "network_name"
}

resource "baremetal_core_internet_gateway" "CompleteIG" {
    compartment_id = "${var.compartment_id}"
    display_name = "CompleteIG"
    vcn_id = "${baremetal_core_virtual_network.t.id}"
}

resource "baremetal_core_route_table" "RouteForComplete" {
    compartment_id = "${var.compartment_id}"
    vcn_id = "${baremetal_core_virtual_network.t.id}"
    display_name = "RouteTableForComplete"
    route_rules {
        cidr_block = "0.0.0.0/0"
        network_entity_id = "${baremetal_core_internet_gateway.CompleteIG.id}"
    }
}

resource "baremetal_core_security_list" "WebSubnet" {
    compartment_id = "${var.compartment_id}"
    display_name = "Public"
    vcn_id = "${baremetal_core_virtual_network.t.id}"
    egress_security_rules = [{
        destination = "0.0.0.0/0"
        protocol = "6"
    }]
    ingress_security_rules = [{
        tcp_options {
            "max" = 80
            "min" = 80
        }
        protocol = "6"
        source = "0.0.0.0/0"
    },
	{
	protocol = "6"
	source = "10.0.0.0/16"
    }]
}

resource "baremetal_core_subnet" "WebSubnetAD1" {
  availability_domain = "${lookup(data.baremetal_identity_availability_domains.ADs.availability_domains[0],"name")}"
  cidr_block = "10.0.1.0/24"
  display_name = "WebSubnetAD1"
  compartment_id = "${var.compartment_id}"
  vcn_id = "${baremetal_core_virtual_network.t.id}"
  route_table_id = "${baremetal_core_route_table.RouteForComplete.id}"
  security_list_ids = ["${baremetal_core_security_list.WebSubnet.id}"]
}

variable "DBNodeShape" {
    default = "BM.DenseIO1.36"
}

variable "CPUCoreCount" {
    default = "2"
}

variable "DBEdition" {
    default = "ENTERPRISE_EDITION"
}

variable "DBAdminPassword" {
    default = "BEstrO0ng_#11"
}

variable "DBName" {
    default = "aTFdb"
}

variable "DBVersion" {
    default = "12.1.0.2"
}


variable "DBDiskRedundancy" {
    default = "HIGH"
}

variable "DBNodeDisplayName" {
    default = "MyTFDatabaseNode0"
}

variable "DBNodeDomainName" {
    default = "mycompany.com"
}

variable "DBNodeHostName" {
    default = "myOracleDB"
}

resource "baremetal_database_db_system" "t" {
  availability_domain = "${data.baremetal_identity_availability_domains.ADs.availability_domains.0.name}"
  compartment_id = "${var.compartment_id}"
  cpu_core_count = "${var.CPUCoreCount}"
  database_edition = "${var.DBEdition}"
  db_home {
    database {
      "admin_password" = "${var.DBAdminPassword}"
      "db_name" = "${var.DBName}"
    }
    db_version = "${var.DBVersion}"
    display_name = "MyTFDB"
  }
  disk_redundancy = "NORMAL"
  shape = "${var.DBNodeShape}"
  subnet_id = "${baremetal_core_subnet.WebSubnetAD1.id}"
  ssh_public_keys = ["${var.ssh_public_key}"]
  display_name = "MyTFDatabaseNode0"
  domain = "${var.DBNodeDomainName}"
  hostname = "${var.DBNodeHostName}"
}

data "baremetal_database_db_nodes" "DBNodes" {
    compartment_id = "${var.compartment_id}"
    db_system_id = "${baremetal_database_db_system.t.id}"
}

data "baremetal_database_db_node" "DBNodeDetails" {
   db_node_id = "${data.baremetal_database_db_nodes.DBNodes.db_nodes.0.id}"
}

data "baremetal_core_vnic" "DBNodeVnic" {
    vnic_id = "${data.baremetal_database_db_node.DBNodeDetails.vnic_id}"
}
```

**Comment by mauropagano** [comment]
@mikejihbe makes perfect sense to me, so the solution (on your side) is just to change the example with a working one (the code I uploaded was based on the example code) or is there going to be a fix from your side? Asking based on @josephholsten comment above

**Comment by mikejihbe** [comment]
Yes, I think this is just a documentation issue that we should resolve.

**Comment by craigcarl-oracle** [comment]
I'll get a new example up this week.

**Comment by codycushing** [comment]
linked #149 
@rcohenma please update the example with your dbaas updates

**Comment by rcohenma** [comment]
The example was fixed with v1.0.15
